velox/external/utf8proc/utf8procImpl.h

```diff
@@ -1007,6 +1007,103 @@ UTF8PROC_DLLEXPORT int utf8proc_codepoint_length(utf8proc_int32_t uc) {
     return 0;
 }
 
+/// This function is not part of the original utf8proc.
+/// Tries to get the length of UTF-8 encoded code point.  A
+/// positive return value means the UTF-8 sequence is valid, and
+/// the result is the length of the code point.  A negative return value means
+/// the UTF-8 sequence at the position is invalid, and the length of the invalid
+/// sequence is the absolute value of the result.
+///
+/// @param input Pointer to the first byte of the code point.
+/// @param size Number of available bytes.
+/// @return the length of the code point or negative the number of bytes in the
+/// invalid UTF-8 sequence.
+///
+/// Adapted from tryGetCodePointAt in
+/// https://github.com/airlift/slice/blob/master/src/main/java/io/airlift/slice/SliceUtf8.java
+UTF8PROC_DLLEXPORT int32_t
+utf8proc_try_char_length(const char* input, uint32_t size) {
+  auto charLength = utf8proc_char_length(input);
```

Comment:
will this crash if input is nullptr or size is zero?

Reply:
Yes, it will. Should I document that input must not be null and size should be > 0 and there must be at least 'size' valid bytes in input?